src/main/java/nonapi/io/github/classgraph/fastzipfilereader/LogicalZipFile.java

```diff
@@ -609,11 +613,51 @@ private void readCentralDirectory(final ZipFileSliceReader zipFileSliceReader, f
                 }
                 final boolean isDeflated = compressionMethod == /* deflated */ 8;
 
+                final int lastModifiedTime = entryBytes != null ? ZipFileSliceReader.getShort(entryBytes, entOff + 12)
+                        : zipFileSliceReader.getInt(cenPos + entOff + 12);
```

Comment:
You've got `getShort` followed by `getInt` here -- this is probably incorrect, since the `entryBytes != null` test is just to test whether to use memory-mapped IO or not. There are several other cases of this below.

Reply:
Good catch.  Fixing now.